uxarray/grid/grid.py

```diff
@@ -223,19 +238,15 @@ def __eq__(self, other) -> bool:
         If two grids are equal : bool
         """
 
-        if other is not None:
-            # Iterate over dict to set access attributes
-            for key, value in self._ugrid_mapping.items():
-                # Check if all grid variables are equal
-                if self._ds.data_vars is not None:
-                    if value in self._ds.data_vars:
-                        if not self._ds[value].equals(other._ds[key]):
-                            return False
-        else:
+        # TODO: fix failing case where grids are the same, but one has more variables constructed
+        try:
+            if self._ds == other._ds:
+                return True
+            else:
+                return False
+        except:
             return False
```

Comment:
@erogluorhan 

I'd like to hear any suggestions on how to update our `==` operator after the removal of `grid_var_names`. Any suggestions?

Reply:
Great question! I think you are concerned about this scenario, right: We get a UGRID input with custom dim names and because of our standardization of those names into our UGRID dim names, they no longer have the same dim names. 

However, regardless from the removal of `grid_var_names `, I think we would return the same result, i.e. `False`, for this scenario. Because comparing `self._ds` to `other._ds`, they'd have different dim names. Am I missing something here?

Considering how to fix this though, I don't have an immediate answer, but preliminary thought: We'd want this scenario to return `True` instead, right? If so, we'd need to check every single dim & var of each of these grids and see if their values are equal. regardless of their names.

Reply:
> Great question! I think you are concerned about this scenario, right: We get a UGRID input with custom dim names and because of our standardization of those names into our UGRID dim names, they no longer have the same dim names.

That scenario is actually not possible, since we should never have any mismatched dimension or variable names now

Reply:
Oh, I see it now, yes! However, any suggestions will be depending on your thoughts for the following quote from an above topic:

>  Problem: It will have our standardized names showing up; however, the user expects to see the original ugrid dataset's custom names because the flow was simply: read in ugrid, encode as ugrid; why variable/dim name changes?
          - To address this, we'd still need to keep an internal `grid_var_names`-like dict. Tho

Reply:
> Oh, I see it now, yes! However, any suggestions will be depending on your thoughts for the following quote from an above topic:

I've pushed a few changes that should cover the case of grids being equal. (a grid with the same coordinates and face node connectivity can be thought of as equivalent, unless we want to check for more in this operator)